share/revng/test/configuration/revng-qa/for-data-layout-analysis.yml

```diff
@@ -0,0 +1,21 @@
+tags:
+  - name: for-data-layout-analysis
+    variables:
+      CFLAGS:
+        - -O2
+        - -Wno-pointer-bool-conversion
+        - -fno-unroll-loops
+        - -fno-inline
+        - -fno-optimize-sibling-calls
+        - -fno-stack-protector
+sources:
+  - tags: [for-data-layout-analysis]
+    members:
+      - share/revng/test/tests/analysis/data-layout-analysis/linked-lists.c
+      - share/revng/test/tests/analysis/data-layout-analysis/segments-and-sections.c
+commands:
+  - type: revng-qa.compiled-for-data-layout-analysis
+    from:
+      - type: source
+        filter: for-data-layout-analysis
+    command: clang $INPUT -o $OUTPUT $CFLAGS
```

Comment:
1. Given that we're not cross-compiling, we could move all of this to `revng-c` directly.

   The reason to put stuff into `revng-qa` is to avoid devs from downloading all the cross-toolchains when running `test/revng` and `test/revng-c`.

   In fact, I'd collapse this into a single command in `test/revng-c` that compiles, decompiles and does whatever it needs to do.
   Same for `for-pretty-int-formatting.yml`.
2. Add a comment on the reasons why we need `clang` and why we want the `CFLAGS` above. We'll forget in 3, 2, 1...